src/main/webapp/features/shared/services/game/GameModel.service.js

```diff
@@ -0,0 +1,95 @@
+'use strict';
+
+angular.module('catan')
+        .factory('GameModel', ['$q', 'Remote', 'User', '$timeout', function ($q, Remote, User, $timeout) {
+
+            var GAME_PRIMARY_KEY = "gameId";
+
+            var STATUS_NOT_LOADED = 0,
+                STATUS_LOADED = 1;
+
+            return function(idOrDetails) {
+
+                var id,
+                    status = STATUS_NOT_LOADED,
+                    details = {},
+                    updatingTimeout = null;
+
+                if (typeof idOrDetails === "number") {
+                    id = idOrDetails;
+                } else if (typeof idOrDetails === "object") {
+                    id = idOrDetails[GAME_PRIMARY_KEY];
+                    details = idOrDetails;
+                    status = STATUS_LOADED;
+                }
+
+                return {
+                    isLoaded: function() {
+                        return status === STATUS_LOADED;
+                    },
+                    load: function() {
+                        var deferred = $q.defer();
+
+                        Remote.game.details({gameId: id}).then(function(response) {
+                            details = response.data;
+                            status = STATUS_LOADED;
+                            deferred.resolve();
+                        }, function() {
+                            deferred.reject();
+                        });
+
+                        return deferred.promise;
+                    },
+                    startUpdating: function(delay, onEverySuccess, onEveryError) {
+                        var self = this;
+                        this.stopUpdating();
```

Comment:
I mean, why we call this.stopUpdating() inside  startUpdating function? as I understand, we should do it only when startUpdating function was already called and we should cancel previously created timer. so the question is it really can happen, when we call startUpdating more than one time?